Mathlib/Combinatorics/SimpleGraph/Matching.lean

```diff
@@ -226,4 +226,37 @@ lemma odd_matches_node_outside {u : Set V} {c : ConnectedComponent (Subgraph.del
 
 end Finite
 end ConnectedComponent
+
+/--
+A graph is matching free if it has no perfect matching. It does not make much sense to
+consider a graph being free of just matchings, because any non-trivial graph has those.
+-/
+def IsMatchingFree (G : SimpleGraph V) := ∀ M : Subgraph G, ¬ M.IsPerfectMatching
+
+lemma IsMatchingFree.mono {G G' : SimpleGraph V} (h : G ≤ G') (hmf : G'.IsMatchingFree) :
+    G.IsMatchingFree := by
+  intro x
+  by_contra! hc
+  apply hmf (x.map (SimpleGraph.Hom.ofLE h))
+  refine ⟨hc.1.map_ofLE h, ?_⟩
+  intro v
+  simp only [Subgraph.map_verts, Hom.coe_ofLE, id_eq, Set.image_id']
+  exact hc.2 v
+
+lemma Fintype.exists_maximal {α} [Fintype α] [PartialOrder α] {a : α} {p : α → Prop}
+    (h : p a) : ∃ b : α, a ≤ b ∧ p b ∧ ∀ c, c > b → ¬ p c := by
```

Comment:
@apnelson1, thoughts?

Reply:
This should probably be phrased with `Maximal p b` (from #14721) as the conclusion. The particular gt -> not version in the statement above is `Maximal.not_prop_of_lt`.